sdk/openai/openai/MIGRATION.md

```diff
@@ -0,0 +1,498 @@
+
+### Audio transcription
+
+The following example shows how to migrate the `getAudioTranscription` method call.
+
+Original code:
+```typescript
+import { readFile } from "fs/promises";
+
+const audioFilePath = "path/to/audio/file";
+const audio = await readFile(audioFilePath);
+const result = await client.getAudioTranscription(deploymentName, audio);
+```
+
+Migrated code:
+```typescript
+import { createReadStream } from "fs";
+
+const result = await client.audio.transcriptions.create({
+  model: '',
+  file: createReadStream(audioFilePath),
+});
+```
+
+Notice that:
+- The `getAudioTranscription` method has been replaced with the `audio.transcriptions.create` method
+- The `AzureOpenAI` has to be constructed with the `deployment` option set to the deployment name in order to use audio operations such as `audio.transcriptions.create`
+- The `model` property is required to be set in the options object but its value is not used in the operation so feel free to set it to any value
+- The `file` property accepts a variety of types including `Buffer`, `fs.ReadaStream`, and `Blob` but in this example, a file is streamed from disk using `fs.createReadStream`
+
+### Audio translation
+
+The following example shows how to migrate the `getAudioTranslation` method call.
+
+Original code:
+```typescript
+import { readFile } from "fs/promises";
+
+const audioFilePath = "path/to/audio/file";
+const audio = await readFile(audioFilePath);
+const result = await client.getAudioTranslation(deploymentName, audio);
+```
+
+Migrated code:
+```typescript
+import { createReadStream } from "fs";
+
+const result = await client.audio.translations.create({
+  model: '',
+  file: createReadStream(audioFilePath),
+});
+```
+
+Notice that:
+- The `getAudioTranslation` method has been replaced with the `audio.translations.create` method
+- All other changes are the same as in the audio transcription example
+
+### Assistants
+
+The following examples show how to migrate some of the `AssistantsClient` methods.
+
+#### Assistant creation
+
+Original code:
+```typescript
+const options = {
+  model: azureOpenAIDeployment,
+  name: "Math Tutor",
+  instructions:
+    "You are a personal math tutor. Write and run JavaScript code to answer math questions.",
+  tools: [{ type: "code_interpreter" }],
+};
+const assistantResponse = await assistantsClient.createAssistant(options);
+```
+
+Migrated code:
+```typescript
+const options = ...;
+const assistantResponse = await assistantsClient.beta.assistants.create(
+  options
+);
+```
+
+Notice that:
+- The `createAssistant` method has been replaced with the `beta.assistants.create` method
+- The names of the properties in the `options` object are the same as in the Azure OpenAI service API, following the snake case convention instead of the camel case convention used in the `AssistantsClient`. This is true for all the properties across all requests and responses in the `AzureOpenAI` client
```

Comment:
this isn't exemplified above. Intentional?

Reply:
Ah good point, moved it to the chat completion example.